compiler/packages/babel-plugin-react-compiler/src/Inference/AnalyseFunctions.ts

```diff
@@ -110,84 +71,39 @@ function lower(func: HIRFunction): void {
   inferMutableRanges(func);
   rewriteInstructionKindsBasedOnReassignment(func);
   inferReactiveScopeVariables(func);
-  inferMutableContextVariables(func);
   func.env.logger?.debugLogIRs?.({
     kind: 'hir',
     name: 'AnalyseFunction (inner)',
     value: func,
   });
 }
 
-function infer(
-  loweredFunc: LoweredFunction,
-  state: IdentifierState,
-  context: Array<Place>,
-): void {
-  const mutations = new Map<string, Effect>();
+function infer(loweredFunc: LoweredFunction): void {
+  const knownMutated = inferMutableContextVariables(loweredFunc.func);
   for (const operand of loweredFunc.func.context) {
-    if (
-      isMutatedOrReassigned(operand.identifier) &&
-      operand.identifier.name !== null
-    ) {
-      mutations.set(operand.identifier.name.value, operand.effect);
-    }
-  }
-
-  for (const dep of loweredFunc.dependencies) {
-    let name: IdentifierName | null = null;
-
-    if (state.properties.has(dep.identifier)) {
-      const receiver = state.properties.get(dep.identifier)!;
-      name = receiver.identifier.name;
-    } else {
-      name = dep.identifier.name;
-    }
-
-    if (isRefOrRefValue(dep.identifier)) {
+    const identifier = operand.identifier;
+    CompilerError.invariant(operand.effect === Effect.Unknown, {
+      reason: 'Unexpected unknown effect',
```

Comment:
we expect the effect to be unknown in the condition, but the error message says the opposite (?)